### PR TITLE
Add saveResrouce Test

### LINKED ISSUE #2263: Write feature tests for saving a resource
Including:
* For new resource (first save)
* For subsequent saves
* When resource is not valid
* When resource is unchanged

### PR DESCRIPTION
**Pull request by aaron-collier**
## Why was this change made?

Part of #2263 -- adds a test for saving a resource. 

## How was this change tested?

Feature test

## Which documentation and/or configurations were updated?

N/A

### CODE REVIEW

File: __tests__/feature/saveResource.test.js [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+      expect(screen.getByRole('button', { name: 'Change language for foo' })).toHaveTextContent('Language: English')
+      // Input is Empty
+      expect(input).toHaveValue('')
+
+      expect(saveBtn).not.toBeDisabled()
+      fireEvent.click(saveBtn)
+
+      const modalSave = screen.getByRole('button', { name: 'Save Group' })
+      fireEvent.click(modalSave)
+      expect(screen.findByText('URI for this resource'))
```
[the remark this answers is not in the document]
Comment (jgreben) [reply __tests__/feature/saveResource.test.js:49]: @aaron-collier I was looking at this a little more, and to my mind--this being a feature test--maybe we do not care at this point if we get a URI, only that the group chooser modal appears and that the save button is clickable. Maybe it's more important just to check that there is a dropdown list of the various group institutions?

File: __tests__/feature/saveResource.test.js [hunk 2]
```diff
@@ -1,7 +1,8 @@
-import { renderApp, createHistory } from 'testUtils'
+import { renderApp, createHistory, createStore } from 'testUtils'
 import { fireEvent, wait, screen } from '@testing-library/react'
 import * as sinopiaServer from 'sinopiaServer'
 import { getFixtureResourceTemplate, rtFixturesGroups } from 'fixtureLoaderHelper'
+import { createState } from '../testUtilities/stateUtils'
```
Comment (jgreben) [inline __tests__/feature/saveResource.test.js:5]: I don't think we need this anymore

File: __tests__/feature/saveResource.test.js [hunk 3]
```diff
@@ -15,7 +16,9 @@ describe('saving a resource', () => {
     const history = createHistory(['/editor/resourceTemplate:bf2:Title:Note'])
 
     it('opens the resource template', async () => {
-      renderApp(null, history)
+      const state = createState({ buildResourceWithLiteral: true })
+      const store = createStore(state)
```
Comment (jgreben) [inline __tests__/feature/saveResource.test.js:20]: I don't think we need these anymore

**Comment by aaron-collier** [comment]
@jgreben good catch. done.

**Pull request merged**